### PR TITLE
Add support for RSHIM under FreeBSD.

### PR DESCRIPTION
**Pull request by hselasky**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,6 +15,7 @@ CDEF = -D_FILE_OFFSET_BITS=64 -DRSHIM_LOG_ENABLE=1 -DFUSE_USE_VERSION=30
 CDEF += -DHAVE_RSHIM_NET
 CDEF += -DHAVE_RSHIM_PCIE
 CDEF += -DHAVE_RSHIM_PCIE_LF
+CDEF += -DHAVE_RSHIM_FUSE
```
Comment (lsun100) [inline Makefile]: Use "HAVE_RSHIM_FUSE" for Linux and "HAVE_RSHIM_CUSE" for FreeBSD seems a little confusing. 
Linux code actually uses both fuse and cuse. Probably use something else for this difference?

File: rshim.c [hunk 2]
```diff
@@ -2132,7 +2799,13 @@ static int rshim_fs_init(struct rshim_backend *bd)
   char *argv[] = {"./bfrshim", "-f"};
   int i, rc;
 
+#ifdef HAVE_RSHIM_CUSE
+  if (cuse_init() != CUSE_ERR_NONE)
+    return -1;
+#endif
+
   for (i = 0; i < RSH_DEV_TYPES; i++) {
+#ifdef FUSE
```
Comment (lsun100) [inline rshim.c]: Is it a typo here?  Linux doesn't have this definition.

Comment (hselasky) [reply rshim.c]: Yes, this is a typo.

Comment (hselasky) [reply rshim.c]: should be fixed now.

File: Makefile (hunk 1 above)
Comment (hselasky) [reply Makefile]: ifdef __linux__ and ifdef __FreeBSD__ is not the right thing for this. Do you have any suggestions?

Comment (lsun100) [reply Makefile]: I don't have good suggestions either.. Let's make it working first. We could consider improving it later, such as isolating the fs implementation into separate files...

File: rshim.c (hunk 2 above)
Comment (lsun100) [reply rshim.c]: Other code looks ok to me. I'll try to verify it in Linux today.

**Pull request merged**